graphql/middleware.go

```diff
@@ -0,0 +1,36 @@
+package graphql
+
+import "context"
+
+type ComputationInput struct {
+	Id          string
+	Query       string
+	ParsedQuery *Query
+	Variables   map[string]interface{}
+	Ctx         context.Context
+	Previous    interface{}
+}
+
+type ComputationOutput struct {
+	Metadata map[string]interface{}
+	Current  interface{}
+	Error    error
+}
+
+type MiddlewareFunc func(input *ComputationInput, output *ComputationOutput, next NextFunc)
+type NextFunc func(input *ComputationInput, output *ComputationOutput)
+
+func _runMiddlewares(index int, middlewares []MiddlewareFunc, input *ComputationInput, output *ComputationOutput) {
```

Comment:
This would be cleaner as a lambda function below:
```go
func runMiddlewares(middlewares []MiddlewareFunc, input *ComputationInput, output *ComputationOutput) {
    var run func(index int, input *ComputationInput, output *ComputationOutput)
    run = func(index int, input *ComputationInput, output *ComputationOutput) {
        if index >= len(middlewares) {
            return
        }
        middleware := middlewares[index]
        middle(input, output, func(input *ComputationInput, output *ComputationOutput) {
              run(index+1, input, output)
        }
    }
    run(0, input, output)
}